libs/coin-modules/coin-ton/src/transaction.ts

```diff
@@ -28,6 +29,109 @@ SEND ${
 }
 TO ${recipient}`;
 
+const fromTransactionPayloadRaw = (payload: TonPayloadFormatRaw): TonPayloadFormat => {
+  switch (payload.type) {
+    case "unsafe":
+      return {
+        type: payload.type,
+        message: Cell.fromBase64(payload.message),
+      };
+    case "jetton-transfer":
+      return {
+        type: payload.type,
+        queryId: payload.queryId ? BigInt(payload.queryId) : null,
+        amount: BigInt(payload.amount),
+        destination: Address.parse(payload.destination),
+        responseDestination: Address.parse(payload.responseDestination),
+        customPayload: payload.customPayload ? Cell.fromBase64(payload.customPayload) : null,
+        forwardAmount: BigInt(payload.forwardAmount),
+        forwardPayload: payload.forwardPayload ? Cell.fromBase64(payload.forwardPayload) : null,
+        knownJetton: payload.knownJetton,
+      };
+    case "nft-transfer":
+      return {
+        type: payload.type,
+        queryId: payload.queryId ? BigInt(payload.queryId) : null,
+        newOwner: Address.parse(payload.newOwner),
+        responseDestination: Address.parse(payload.responseDestination),
+        customPayload: payload.customPayload ? Cell.fromBase64(payload.customPayload) : null,
+        forwardAmount: BigInt(payload.forwardAmount),
+        forwardPayload: payload.forwardPayload ? Cell.fromBase64(payload.forwardPayload) : null,
+      };
+    case "jetton-burn":
+      return {
+        type: payload.type,
+        queryId: payload.queryId ? BigInt(payload.queryId) : null,
+        amount: BigInt(payload.amount),
+        responseDestination: Address.parse(payload.responseDestination),
+        // Handle customPayload could be a buffer, we should try/catch the fromBase64
+        customPayload: payload.customPayload ? Cell.fromBase64(payload.customPayload) : null,
+      };
+    case "add-whitelist":
+      return {
+        type: payload.type,
+        queryId: payload.queryId ? BigInt(payload.queryId) : null,
+        address: Address.parse(payload.address),
+      };
+    case "single-nominator-withdraw":
+      return {
+        type: payload.type,
+        queryId: payload.queryId ? BigInt(payload.queryId) : null,
+        amount: BigInt(payload.amount),
+      };
+    case "single-nominator-change-validator":
+      return {
```

Comment:
case "single-nominator-change-validator" is the same as case "add-whitelist":

Reply:
Nice catch, I'll update